### PR TITLE
Support Node 12 and refactor TransactionController for EIP-1559 compat

### PR DESCRIPTION
**Pull request by onyb**
|GitHub|https://github.com/brave/brave-browser/issues/17160|
-|-

### Prerequisites

* https://github.com/brave/dnode/pull/1
* https://github.com/brave/eth-hd-keyring/pull/10
* https://github.com/brave/KeyringController/pull/15

### What is this about?

This PR refactors `TransactionController` to use the new `@ethereumjs/tx` package to build raw transactions. It will allow us to create gas-fee-market transactions (EIP-1559) when the London hard-fork is activated. There is no EIP-1559 support or visual changes in this PR.

Also adds Node 12 support.

### :skull_and_crossbones: CAUTION :skull_and_crossbones:    (attn @diracdeltas)

I removed `sesify` (see commit a5648cd) because it was not supported on Node 12 (necessary for EIP-1559). Metamask had removed it long back for the same reason, but introduced another library called [lavamoat](https://github.com/LavaMoat/LavaMoat) as a replacement. My understanding is that it makes things more secure, but would like an expert opinion on it. I created a security review ticket to evaluate this commit: https://github.com/brave/security/issues/535

### CODE REVIEW

**Review by bbondy — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -87,14 +87,16 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.5.5",
-    "@brave/eth-keyring-controller": "^5.4.3",
+    "@brave/eth-keyring-controller": "onyb/KeyringController#upgrade-packages",
```
Comment (diracdeltas) [inline package.json:90]: could you please open a PR in the brave repo? we prefer not to depend on repos in people's personal github accounts if possible

File: package.json [hunk 2]
```diff
@@ -118,23 +120,22 @@
     "d3": "^5.15.0",
     "debounce-stream": "^2.0.0",
     "deep-freeze-strict": "1.1.1",
-    "dnode": "^1.2.2",
+    "dnode": "brave/dnode#remove-weak-package",
```
Comment (diracdeltas) [inline package.json:123]: could you address this dependabot alert? https://github.com/brave/dnode/security/dependabot

File: package.json (hunk 1 above)
Comment (diracdeltas) [reply package.json:90]: looks like the brave repo may have been updated already so this just needs to point to that

**Review by diracdeltas — CHANGES_REQUESTED** [review]
see comments above + https://github.com/brave/security/issues/535#issuecomment-893874494

File: package.json (hunk 1 above)
Comment (onyb) [reply package.json]: Fixed.

File: package.json (hunk 2 above)
Comment (onyb) [reply package.json]: Done.

**Comment by onyb** [comment]
@diracdeltas Dropped the commit that removed sesify. See my reply in the security review: https://github.com/brave/security/issues/535#issuecomment-894235773

**Comment by onyb** [comment]
Going to merge this PR to `eip1559` branch. We'll keep using SESify for now, which maintains the status-quo, so brave/security#535 is no longer a blocker for merging.

**Pull request merged**